### DISCUSSION TITLE
Reviewing the Touch to Stop Speech

### DISCUSSION THREAD

**Discussion by Irineu333** [Irineu333/SpeakTouch#132] [Ideas]
As evidenced by @PatrykMis a few months ago, the "Touch to stop speech" [^1] function as currently implemented causes a delay in reading the nodes. Although it is the predominant behavior in screen readers, stopping the reading with just one touch might not be the ideal behavior. I created this discussion to decide how to proceed.

[^1]: #69

**Comment by Irineu333** [comment]
I believe that as a first change we should restrict the "touch to stop speech" function to the node currently in focus. It's redundant to enable this function for other nodes, considering that to start reading them, it is necessary to stop reading the current ones. So, we would have the following behavior: 1) touching a node puts it in focus, which starts its reading; 2) touching a node in focus will stop the reading if it is in progress or read it again if it has finished.

**Reply by Irineu333** [reply]
As an alternative, and even easier to implement, we could simply code the following behavior: 1) touching a node starts its reading; 2) touching a node in focus restarts its reading.

**Reply by Irineu333** [reply]
The implementation of either of the described behaviors closes the following #70.

**Comment by Irineu333** [comment]
I remember @PatrykMis suggesting implementing the function to stop reading on the two-finger touch gesture. We can do that for now, but in the future, I want to implement the pause and resume function, which is usually assigned to this gesture.

**Comment by PatrykMis** [comment]
> As an alternative, and even easier to implement, we could simply code the following behavior: 1) touching a node starts its reading; 2) touching a node in focus restarts its reading.

It's a good approach, as tapping with two fingers will pause reading, so I think we don't need different approach for stopping speech. Further more, touching another or even the same note will not cause delays.
